src/screens/CreateAccountScreen.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from "react";
+import { View, Text, TextInput, StyleSheet } from "react-native";
+import { Button } from "../components/Button";
+import axios from "axios";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { Screens } from "./MainScreen";
+import sha256 from "crypto-js/sha256";
+
+type CreateAccountScreenProps = { setPage: (screen: Screens) => void };
+
+const CreateAccount: React.FC<CreateAccountScreenProps> = ({ setPage }) => {
+	const [errorMsg, setErrorMsg] = useState("");
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+
+	/**
+	 * Validates that the email the user provided has a valid email format
+	 * @param email
+	 * @returns
+	 */
+	const validateEmail = (email: string) => {
+		setErrorMsg("");
+
+		if (!email) {
+			setEmail("");
+			return;
+		}
+
+		const reg: RegExp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/;
+
+		if (reg.test(email) === false) {
+			setErrorMsg("Please provide a valid email");
+			return;
+		} else {
+			setErrorMsg("");
+			setEmail(email);
+			return;
+		}
+	};
```

Comment:
you prob noticed both validate-email functions are the exact same, maybe we should move the regex part of this to a small util function?
or another option (what I would do) make a custom hook which returns `email, setEmail, emailError`. When `setEmail` is run, you could check for regex, and then don't actually set `email` if there's an error, just set `emailError`. That hook could then be tested, run in both places, and cleanup the handling here. Or you could decide that's too much work, i don't have a super strong preference.
Lmk if you want to get on a call to discuss that.

Reply:
Done and dusted! 😄